src/olympia/devhub/tests/test_views_validation.py

```diff
@@ -243,38 +239,52 @@ def test_context_and_content(self):
         assert doc('#upload-addon').attr('data-upload-url-unlisted') == (
             reverse('devhub.standalone_upload_unlisted'))
 
-    @mock.patch('olympia.devhub.tasks.run_validator')
+    @mock.patch('olympia.devhub.tasks.run_addons_linter')
     def test_filename_not_uuidfied(self, validate_mock):
         validate_mock.return_value = json.dumps(amo.VALIDATOR_SKELETON_RESULTS)
         url = reverse('devhub.upload')
-        data = open(get_image_path('animated.png'), 'rb')
-        self.client.post(url, {'upload': data})
+
+        fpath = 'src/olympia/files/fixtures/files/webextension_no_id.xpi'
+
+        with open(fpath) as file_:
+            self.client.post(url, {'upload': file_})
+
         upload = FileUpload.objects.get()
         response = self.client.get(
             reverse('devhub.upload_detail', args=(upload.uuid.hex,)))
-        assert 'Validation Results for animated.png' in response.content
+        assert 'Validation Results for webextension_no_id' in response.content
 
-    @mock.patch('olympia.devhub.tasks.run_validator')
+    @mock.patch('olympia.devhub.tasks.run_addons_linter')
     def test_upload_listed_addon(self, validate_mock):
         """Listed addons are not validated as "self-hosted" addons."""
         validate_mock.return_value = json.dumps(amo.VALIDATOR_SKELETON_RESULTS)
-        self.url = reverse('devhub.upload')
-        data = open(get_image_path('animated.png'), 'rb')
-        self.client.post(self.url, {'upload': data})
-        # Make sure it was called with listed=True.
-        assert validate_mock.call_args[1]['listed']
+        url = reverse('devhub.upload')
+
+        fpath = 'src/olympia/files/fixtures/files/webextension_no_id.xpi'
+
+        with open(fpath) as file_:
```

Comment:
I know there are lots of tests broken in the same way already, and I'm not suggesting to fix those in this PR, but here and below it should be:
```suggestion
        with open(fpath, 'rb') as file_:
```

The person who will end up porting these tests to Python 3 will thank you :)

Reply:
I already have the devhub py3 branch in progress :)
Although I didn't touch `test_views_validation.py` yet because validator.